Tools/AP_Periph/can.cpp

```diff
@@ -1277,6 +1302,41 @@ static void processRx(void)
     }
 }
 
+#ifdef HAL_PERIPH_CAN_MIRROR
+static void processMirror(void)
+{
+    const uint64_t deadline = AP_HAL::native_micros64() + 1000000;
+
+    for (auto &ins : instances) {
+        if (ins.iface == nullptr || ins.mirror_queue == nullptr) { // can't send on a null interface
+            continue;
+        }
+
+        for (uint32_t i = 0; i < ins.mirror_queue->available (); i++) { // limit how long we can loop
+            AP_HAL::CANFrame txmsg {};
+
+            if (!ins.mirror_queue->peek(txmsg)) {
+                break;
+            }
+
+            // FIXME: Do we need to set CANFD flags?
```

Comment:
I don't know if I have to do anything with CANFD packets? It's apparent to me that `processTx()` is doing some work, but I'm not sure what I need to do in this case. Particularly since you can't just fit the same packets from a FD interface to a non FD interface. Is it fine to just declare that this feature only works with equally clocked ports? (It seems a bit nonsenical to run it on mismatched bus speeds).

Reply:
if CANFD is on the way in then use CANFD on the way out

Reply:
Should be fine, remove the comment.